src/api/java/mekanism/api/gear/ModuleData.java

```diff
@@ -22,22 +22,27 @@
 import mekanism.api.gear.config.ModuleBooleanConfig;
 import mekanism.api.gear.config.ModuleConfig;
 import mekanism.api.providers.IItemProvider;
-import mekanism.api.providers.IModuleDataProvider;
 import net.minecraft.Util;
+import net.minecraft.core.Holder;
 import net.minecraft.network.RegistryFriendlyByteBuf;
 import net.minecraft.network.codec.StreamCodec;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 @NothingNullByDefault
-public class ModuleData<MODULE extends ICustomModule<MODULE>> implements IModuleDataProvider<MODULE> {
+@SuppressWarnings("removal")
+public class ModuleData<MODULE extends ICustomModule<MODULE>> implements mekanism.api.providers.IModuleDataProvider<MODULE> {
 
     private final Function<@NotNull IModule<MODULE>, @NotNull MODULE> constructor;
     private final Int2ObjectMap<ConstructedConfigData> configData;
+    @Deprecated(forRemoval = true, since = "10.7.11")
     private final IItemProvider itemProvider;
+    @Nullable//TODO - 1.22: Make this nonnull
```

Comment:
why is it nullable?

Reply:
Because if someone instantiates it via the deprecated `IItemProvider` way, we don't know for certain (given this is during registration) that calling `IItemProvider#asItem` won't cause an exception of some sort to be thrown. I doubt it would, given the module data registry happens after vanilla ones, but I figured just to be safe we were better off just lazily resolving the deprecated way.